wcfsetup/install/files/lib/system/request/RequestHandler.class.php

```diff
@@ -146,8 +148,18 @@ public function handle(string $application = 'wcf', bool $isACPRequest = false):
                     return;
                 }
             } else {
-                \assert($builtRequest instanceof ResponseInterface);
-                $response = $builtRequest;
+                \assert($builtRequest instanceof RequestHandlerInterface);
+
+                $pipeline = new Pipeline([
+                    new HandleStartupErrors(),
+                    new PreventMimeSniffing(),
+                    new AddAcpSecurityHeaders(),
+                    new EnforceCacheControlPrivate(),
+                    new EnforceNoCacheForTemporaryRedirects(),
+                    new EnforceFrameOptions(),
+                ]);
```

Comment:
Aren’t those handlers now registered twice because I don’t see a change to the existing pipeline?

Reply:
The regular middleware pipeline is created within the “then” part of the `if()`, this one is in the “else” part and completely independent.

The `Pipeline` class itself is immutable, thus adding additional middlewares is currently not possible, making a little duplication the easiest solution for now.